webapp/src/components/github_milestone_selector.jsx

```diff
@@ -5,57 +5,42 @@ import React, {PureComponent} from 'react';
 import PropTypes from 'prop-types';
 
 import IssueAttributeSelector from 'components/issue_attribute_selector';
+import Client from 'client';
 
 export default class GithubMilestoneSelector extends PureComponent {
     static propTypes = {
         repo: PropTypes.string.isRequired,
         theme: PropTypes.object.isRequired,
-        selectedMilestone: PropTypes.number.isRequired,
+        selectedMilestone: PropTypes.object.isRequired,
         onChange: PropTypes.func.isRequired,
-        milestones: PropTypes.array,
-        actions: PropTypes.shape({
-            getMilestones: PropTypes.func.isRequired,
-        }).isRequired,
     };
 
     loadMilestones = async () => {
         if (this.props.repo === '') {
             return [];
         }
 
-        const options = await this.props.actions.getMilestones(this.props.repo);
-
-        if (options.error) {
+        try {
+            const options = await Client.getMilestones(this.props.repo) || [];
```

Comment:
We should try to avoid using the `Client` directly in the components. These API calls can be in the `actions.js` file, and passed in through props, even if we do not store the results in redux. @larkox What are your thoughts?

Reply:
It is indeed a standard in our projects to do that that way. Probably it will be better that way.

Reply:
@mickmister @larkox Done: https://github.com/mattermost/mattermost-plugin-github/pull/343/commits/3ba4df8c3c51abe5e7d9d302ff64e3f36f61647e